### DISCUSSION TITLE
REST API Delete user and his local folder (OS dir)

### DISCUSSION THREAD

**Discussion by rnenergy** [drakkan/sftpgo#903] [Q&A] [answered]
Hello. I'll be very grateful for your help!
After deleting a user via the API, his folder remains in the local storage (the default for Debian is /srv/sftpgo/data/"user"). Is it possible to set up the deletion of this folder along with the user through the API?
Best regards, Roman.

**Comment by drakkan** [comment, marked as answer]
Hi,

you can use a custom hook for this. Take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/custom-actions.md#provider-events). 

Deleting the user's home directory may take a long time for local fs (for example if it contains a lot of files). It is not applicable for cloud storages and finally if you delete a user by mistake you can lose important files if SFTPGo automatically delete the home dir

**Reply by rnenergy** [reply]
Thanks a lot.
